src/test/java/com/karankumar/bookproject/backend/service/BookServiceTest.java

```diff
@@ -64,6 +67,38 @@ public void whenTryingToSaveBookWithoutAuthorExpectNoSave() {
         Assertions.assertEquals(0, bookService.count());
     }
 
+    /**
+     * Tests book is not saved with numberOfPages > 23000
+     */
+    @Test
+    public void whenTryingToSaveBookWithMaxNumberPageExceedNoSave() {
+        Book book = new Book("Book without author", new Author("First", "Last"), toRead);
+        book.setNumberOfPages(Integer.MAX_VALUE);
+        try {
+            bookService.save(book);
+        } catch (Exception e)
+        {
+            Assertions.assertTrue(ExceptionUtils.getRootCause(e) instanceof ConstraintViolationException);
+        }
+        Assertions.assertEquals(0, bookService.count());
+    }
```

Comment:
Instead of the catch block do you want to use something like below:

`Throwable exception = Assertions.catchThrowable(() -> bookService.save(book))`

Then you can assert the exception variable for the exception type, would be better if we can assert that error message as well if there is any.